plugins/discovery-ec2/src/main/java/org/elasticsearch/cloud/aws/AwsEc2ServiceImpl.java

```diff
@@ -117,15 +117,15 @@ public long delayBeforeNextRetry(AmazonWebServiceRequest originalRequest,
 
         AWSCredentialsProvider credentials;
 
-        if (key == null && secret == null) {
+        if (Strings.isNullOrEmpty(key) || Strings.isNullOrEmpty(secret)) {
```

Comment:
If the defaults are actually empty string, then we should not be checking/allowing for null. Let the NPE happen: if it occurs, it is an internal bug. Otherwise the bug would be hidden away by this leniency. Also, you changed the logic slightly, because now it is logicl OR instead of AND. I think it should stay AND?